apollo-router/src/plugins/connectors/http_json_transport.rs

```diff
@@ -77,43 +78,72 @@ pub(crate) fn make_request(
         &flat_inputs,
     )?;
 
-    let (json_body, body, apply_to_errors) = if let Some(ref selection) = transport.body {
+    let request = http::Request::builder()
+        .method(transport.method.as_str())
+        .uri(uri.as_str());
+
+    // add the headers and if content-type is specified, we'll check that when constructing the body
+    let (mut request, content_type) = add_headers(
+        request,
+        original_request.supergraph_request.headers(),
+        &transport.headers,
+        &flat_inputs,
+    );
+
+    let is_form_urlencoded = content_type
+        .as_ref()
+        .map(|v| v.to_str().unwrap_or_default())
+        == Some("application/x-www-form-urlencoded");
```

Comment:
Does `http` give us a way to compare the semantic value here? Like, if they wanted to pass some extra data after a `;`, I believe we should ignore that.

Reply:
good question, i'll look around

Reply:
turns out the mime crate handles this well